From 411122d16a5830fd79b3c39afd72163cba531cd9 Mon Sep 17 00:00:00 2001
From: kboeluek <kboeluek@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 30 May 2014 14:52:19 +0000
Subject: [PATCH 0546/1076] MANTIS 0020990 extension;
 Platform_Extension_CUSS_1_2: Periodic/Automatic restart of the application

systools @version 1.0.0.18

- mSysProcess.hpp / mSysProcessW32.cpp:
- added a method in mSysProcess::child. it provides access to the current memory usage of a process

git-svn-id: svn://localhost/SelfServiceCommon/trunk@681 90b65887-3827-0410-9a23-83215b262276
---
 .../Interfaces/inc/systools/mSysProcess.hpp       |  2 ++
 .../Massai/cpp/Systools/src/local_def.cpp         | 10 ++++++++--
 .../Massai/cpp/Systools/src/mSysProcessW32.cpp    | 15 +++++++++++++++
 3 files changed, 25 insertions(+), 2 deletions(-)

diff --git a/SelfServiceCommon/Interfaces/inc/systools/mSysProcess.hpp b/SelfServiceCommon/Interfaces/inc/systools/mSysProcess.hpp
index 1055ed30..91269359 100644
--- a/SelfServiceCommon/Interfaces/inc/systools/mSysProcess.hpp
+++ b/SelfServiceCommon/Interfaces/inc/systools/mSysProcess.hpp
@@ -231,6 +231,8 @@ namespace mSysProcess
       virtual unsigned long getProcessIoUsage() const = 0;
       
       virtual unsigned long getProcessTimeUsage() const = 0;
+      
+      virtual unsigned long getProcessMemoryUsage () const = 0;
 
    }; // class child
 
diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
index ccf2e555..882e1dde 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
@@ -1,6 +1,6 @@
 #include "tools/versionlib.h"
 
-VERSION_ID_DT(systools, 1, 0, 0, 17);
+VERSION_ID_DT(systools, 1, 0, 0, 18);
 
 /*!
 \file
@@ -8,12 +8,18 @@ VERSION_ID_DT(systools, 1, 0, 0, 17);
 
 \page massai_systools.lib Release Notes - massai_systools.lib
 
-@version 1.0.0.17
+@version 1.0.0.18
 
 \section changehistory Change history
 
 massai_systools.lib
 
+\version 1.0.0.18
+\li 30.05.2014
+\li extension - BoKa Mantis 0020990: Automated restarts
+\li - mSysProcess.hpp / mSysProcessW32.cpp
+\li   added a method in mSysProcess::child. it provides access to the current memory usage of a process.
+
 \version 1.0.0.17
 \li 15.05.2014
 \li change - BoKa Mantis 0020990: Automated restarts
diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/mSysProcessW32.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/mSysProcessW32.cpp
index e247d0e1..b303e9a5 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/mSysProcessW32.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/mSysProcessW32.cpp
@@ -371,6 +371,7 @@ namespace
       
       unsigned long getProcessIoUsage() const;
       unsigned long getProcessTimeUsage() const;
+      unsigned long getProcessMemoryUsage() const;
 
     private:
       void openProcess();
@@ -1074,6 +1075,20 @@ namespace
     return sum;
   }
 
+  unsigned long myChild::getProcessMemoryUsage() const
+  {
+    PROCESS_MEMORY_COUNTERS memoryCounters;
+    unsigned long result = 0;
+    if(GetProcessMemoryInfo(M_HandleProcess, &memoryCounters, sizeof(memoryCounters)))
+    {
+      result = memoryCounters.WorkingSetSize;
+    }
+    else
+    {
+      logLastError(M_Logging, "GetProcessMemoryInfo");
+    }
+    return result;
+  }
 } // namespace
 
 
-- 
2.41.0.windows.1

